### PR TITLE
Update local ESLint config to match shared config + apply new rules

### PR DESCRIPTION
**Pull request by fregante**
## What does this PR do?

- Follows https://github.com/pixiebrix/eslint-config-pixiebrix/pull/228
- Follows https://github.com/pixiebrix/pixiebrix-extension/pull/7703

## Checklist

- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -60,67 +69,35 @@ module.exports = {
...
+    "no-restricted-imports": [
+      "error",
+      // If they're not specific to the extension, add them to the shared config instead:
+      // https://github.com/pixiebrix/eslint-config-pixiebrix/blob/main/no-restricted-imports.js
+      extendNoRestrictedImports({
+        patterns: [
+          {
+            group: ["react-shadow/emotion"],
+            message:
+              'Use this instead: import EmotionShadowRoot from "@/components/EmotionShadowRoot"',
```
Comment (fregante) [inline .eslintrc.js:82]: `FulfilledPromise`

https://github.com/pixiebrix/pixiebrix-extension/pull/7751#discussion_r1506035165

File: .eslintrc.js [hunk 2]
```diff
@@ -60,67 +69,35 @@ module.exports = {
...
-        message:
-          "Unless the code is using .then(), calling `.mockResolvedValue(undefined)` is the same as leaving it out",
-        selector:
-          "CallExpression[callee.property.name='mockResolvedValue'][arguments.0.name='undefined'][arguments.0.type='Identifier']",
-      },
       // NOTE: If you add more rules, add the tests to eslint-local-rules/noRestrictedSyntax.ts
     ],
-
-    // We want to have a default case to check for `never`
-    "@typescript-eslint/switch-exhaustiveness-check": [
```
Comment (fregante) [inline .eslintrc.js:111]: Extracted:

- https://github.com/pixiebrix/eslint-config-pixiebrix/pull/231

File: .eslintrc.js [hunk 3]
```diff
@@ -60,67 +69,35 @@ module.exports = {
...
-      "error",
-      {
-        allowDefaultCaseForExhaustiveSwitch: true,
-        requireDefaultForNonUnion: true,
-      },
-    ],
-
-    // Rules that depend on https://github.com/pixiebrix/pixiebrix-extension/issues/775
-    "@typescript-eslint/restrict-template-expressions": [
```
Comment (fregante) [inline .eslintrc.js:120]: Extracted:

- https://github.com/pixiebrix/eslint-config-pixiebrix/pull/221

File: .eslintrc.js [hunk 4]
```diff
@@ -200,6 +177,23 @@ module.exports = {
...
+      files: ["./src/*"],
+      rules: {
+        "no-restricted-imports": [
+          "error",
+          extendNoRestrictedImports({
+            patterns: [
+              {
+                group: ["./*"],
+                message:
+                  'Use root-based imports (`import "@/something"`) instead of relative imports.',
```
Comment (fregante) [inline .eslintrc.js:190]: Moved back here from the shared config. This is an extension-only rule.

- https://github.com/pixiebrix/eslint-config-pixiebrix/pull/230

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -1,4 +1,10 @@
 # https://pre-commit.com/hooks.html
+
+# The hooks can be updated with:
+# pre-commit autoupdate
+#
+# And then run updated hooks against all files with:
+# pre-commit run --all-files
```
Comment (fregante) [inline .pre-commit-config.yaml:7]: Copied from Johnny's comment on the app repo

File: package.json [hunk 6]
```diff
@@ -299,6 +299,9 @@
     "yaml-jest-transform": "^2.0.2",
     "yaml-loader": "^0.8.1"
   },
+  "overrides": {
+    "@typescript-eslint/eslint-plugin": "7.0.2"
```
Comment (fregante) [inline package.json:303]: - https://github.com/xojs/eslint-config-xo-typescript/issues/83

File: src/components/StopPropagation.test.tsx [hunk 7]
```diff
@@ -1,5 +1,3 @@
-/* eslint-disable jsx-a11y/click-events-have-key-events */
-/* eslint-disable jsx-a11y/no-static-element-interactions -- Test only */
```
Comment (fregante) [inline src/components/StopPropagation.test.tsx:2]: - https://github.com/pixiebrix/eslint-config-pixiebrix/pull/222

File: src/extensionConsole/pages/onboarding/SetupPage.test.tsx [hunk 8]
```diff
@@ -109,8 +109,6 @@ describe("SetupPage", () => {
       expect(screen.queryByTestId("loader")).toBeNull();
     });
 
-    screen.debug();
```
Comment (fregante) [inline src/extensionConsole/pages/onboarding/SetupPage.test.tsx:112]: - https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-debugging-utils.md

File: .eslintrc.js [hunk 9]
```diff
@@ -60,67 +69,35 @@ module.exports = {
...
-          "JSXElement[openingElement.name.name='Row'] > JSXText:first-child + JSXElement:nth-last-child(2)",
-      },
-      {
-        message:
-          "Use the `uuid` module instead because crypto.randomUUID is not available in http: contexts",
-        selector: 'MemberExpression > Identifier[name="randomUUID"]',
-      },
+      // If they're not specific to the extension, add them to the shared config instead:
+      // https://github.com/pixiebrix/eslint-config-pixiebrix/blob/main/no-restricted-syntax.js
+      ...require("eslint-config-pixiebrix/no-restricted-syntax"),
```
Comment (fregante) [inline .eslintrc.js:92]: Extracted:

- https://github.com/pixiebrix/eslint-config-pixiebrix/pull/228

File: .eslintrc.js [hunk 10]
```diff
@@ -26,8 +35,8 @@ module.exports = {
     "local-rules/noInvalidDataTestId": "error",
     "local-rules/noExpressionLiterals": "error",
     "local-rules/notBothLabelAndLockableProps": "error",
-    "local-rules/preferNullish": "warn",
-    "local-rules/preferNullishable": "warn",
+    "local-rules/preferNullish": "error",
+    "local-rules/preferNullishable": "error",
```
Comment (fregante) [inline .eslintrc.js:39]: There were only a handful of errors. Fixed all

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**